src/mtev_http.c

```diff
@@ -981,6 +1008,13 @@ mtev_http_complete_request(mtev_http_session_ctx *ctx, int mask) {
     if(len <= 0) goto full_error;
     if(len > 0) in->size += len;
     rv = mtev_http_request_finalize(&ctx->req, &err);
+    /* walk the bchain and set the compression */
+    struct bchain *x = ctx->req.first_input;
+    while (x) {
+      x->compression = request_compression_type(&ctx->req);
+      ctx->req.content_length_read += x->size;
```

Comment:
this doesn't look right - I think it's updating the `content_length_read` multiple times for the same `bchain`, in the case we enter this routine multiple times... plus, I think it's increasing the content_length_read based on header bchains, not the payload?